### PR TITLE
Warn on duplicate imports

### LINKED ISSUE #1390: Warn on duplicate imports
Conditions for this:

- [x] Two whole-module imports of the same module
- [x] A whole-module import and a partial-import of the same module
- [x] Overlapping items in two partial-imports of the same module (suggest combining the list)
- [x] Two qualified imports of the same module

### PR DESCRIPTION
**Pull request by garyb**
Resolves #1390

### CODE REVIEW

File: src/Language/PureScript/Sugar/Names/Imports.hs [hunk 1]
```diff
@@ -70,9 +70,36 @@ findImports = foldM (go Nothing) M.empty
...
+              maybe id warnWithPosition pos $
+                tell $ errorMessage $ DuplicateSelectiveImport mn
+
+    let scope = M.insert currentModule [(Nothing, Implicit, Nothing)] imports
     foldM (resolveModuleImport currentModule env) nullImports (M.toList scope)
 
+  where
+  selfCartesianSubset :: [a] -> [(a, a)]
+  selfCartesianSubset (x : xs) = [(x, y) | y <- xs] ++ selfCartesianSubset xs
+  selfCartesianSubset [] = []
```
Comment (garyb) [inline src/Language/PureScript/Sugar/Names/Imports.hs]: Surely this must be a thing that exists elsewhere?

**Comment by garyb** [comment]
When it comes to qualified imports it only warns if there is an exact duplicate match. There is at least one valid reasons for why a module might be imported qualified more than once (importing into pseudo-module for exports as well as an import that is used within in the module somewhere).

File: src/Language/PureScript/Sugar/Names/Imports.hs (hunk 1 above)
Comment (paf31) [inline src/Language/PureScript/Sugar/Names/Imports.hs]: `replicateM 2` is pretty close, but not the same since it returns `[[a]]`.

**Comment by garyb** [comment]
Oops, will fix whatever is wrong first :smile:

**Comment by garyb** [comment]
Ok, that should do it.

**Comment by paf31** [comment]
:+1: Looks nice, thanks!

**Pull request merged**